### ISSUE TITLE
Please document how to serve real credentials in the mocked data

### ISSUE THREAD

**Issue by joshtriplett** [aws/amazon-ec2-metadata-mock#184]
For the purposes of testing an application relying on the IMDS metadata service, I'd like to be able to serve valid AWS credentials via the mocked IMDS.

I'd love to have documentation for how to translate `~/.aws/credentials` into the relevant pieces of the mock configuration, or better yet, an option to serve up a particular profile from those credentials as the mocked data.

**Comment by slimm609** [comment]
@joshtriplett I actually have a blog post about to be published doing exactly this with real credentials and responds the same way as an ec2 instance (all the SDKs pick it up as native instance profile).  I am putting the finishing touches on it and should have it done in the next day or so.  

Will share it once complete

**Comment by joshtriplett** [comment]
@slimm609 That sounds great! Thank you.

**Comment by brycahta** [comment]
@slimm609 That's a great use case! Would you be open to adding to our [Community Use Cases](https://github.com/aws/amazon-ec2-metadata-mock/blob/main/docs/usage.md#community-use-cases) section with the content? Or would you mind if someone from our team added a section if you don't have bandwidth? 

In either case, we would attribute you for the content, but I'm reluctant to simply link the blog post because of the overhead for our team to maintain the integrity of said links.

**Comment by joshtriplett** [comment]
Still here, still interested.

**Comment by jweyrich** [comment]
Speaking on behalf of the company I work for, we're very interested in this too.

**Comment by slimm609** [comment]
@jweyrich DId you look at the blog post above?  There are several things that are outside the scope of what this tool would do (creating loopback addresses, templated configs, etc).  I created a repo that wraps around ec2-metadata-mock to wrap it all together and works natively on the system with the AWS SDKs.  Its in the blog post above as well but here is the repo https://github.com/slimm609/mock-instance-profile

**Comment by jweyrich** [comment]
@slimm609 I don't think a loopback virtual interface is necessary. See [A Guide to Locally Testing Containers with Amazon ECS Local Endpoints and Docker Compose](https://aws.amazon.com/blogs/compute/a-guide-to-locally-testing-containers-with-amazon-ecs-local-endpoints-and-docker-compose/). We define a Docker bridge network in the same CIDR as the IMDS and change the default metadata service endpoint via env var `AWS_EC2_METADATA_SERVICE_ENDPOINT` in the application container itself. The new endpoint has to comply with the IMDS - we tested the 2 services I listed above, but they do not work with the aws-sdk-v3, only with v2. I'm trying to figure out the reason, or find a IMDS service that would work with both versions of the aws-sdk (v2 and v3).
